### PR TITLE
Added VWAP and Hamming moving average indicators

### PR DESCRIPTION
**Pull request by Bapi-Reddy**
I am new to Julia, lmk if you have any suggestions or find issue with my pr.

### CODE REVIEW

File: src/ma.jl [hunk 1]
```diff
@@ -395,3 +395,34 @@ function vwma(cv::Matrix{T}, n::Int64=10)::Array{Float64} where {T<:Real}
...
+```
+
+volume weighted average price (VWAP)
+"""
+function vwap(cv::Matrix{T})::Array{Float64} where {T<:Real}
+    out = zeros(size(cv))[1]
+    close_price = cv[:,1]
+    volume = cv[:,2]
+    out = cumsum(close_price .* volume) ./ cumsum(volume)
+    end
```
Comment (JTAmos) [inline src/ma.jl]: What should this `end` correspond to? Results in precompilation error. Delete.

File: src/ma.jl [hunk 2]
```diff
@@ -395,3 +395,34 @@ function vwma(cv::Matrix{T}, n::Int64=10)::Array{Float64} where {T<:Real}
     end
     return out
 end
+
+"""
+```
+vwap(cv::Matrix{T})::Array{T}
+```
+
+volume weighted average price (VWAP)
```
Comment (JTAmos) [inline src/ma.jl]: Sentence case per convention with other docstrings.

File: src/ma.jl [hunk 3]
```diff
@@ -395,3 +395,34 @@ function vwma(cv::Matrix{T}, n::Int64=10)::Array{Float64} where {T<:Real}
...
+    end
+    return out
+end
+
+"""
+```
+hama(x::Array{T})::Array{T}
+```
+
+hamming moving average (HAMA)
```
Comment (JTAmos) [inline src/ma.jl]: Sentence case per convention with other docstrings.

File: src/ma.jl [hunk 4]
```diff
@@ -395,3 +395,34 @@ function vwma(cv::Matrix{T}, n::Int64=10)::Array{Float64} where {T<:Real}
...
 end
+
+"""
+```
+vwap(cv::Matrix{T})::Array{T}
+```
+
+volume weighted average price (VWAP)
+"""
+function vwap(cv::Matrix{T})::Array{Float64} where {T<:Real}
```
Comment (JTAmos) [inline src/ma.jl:407]: VWAP is generally calculated on intraday prices. Given the absence of timestamps in the function signature, this should probably have some kind of lookback period (e.g. `n::Int=10`).

This appears to be "cumulative" VWMA by default, which is fine as an option but differs from the current convention used by the other functions in this file. I would add another argument after the `n::Int` argument to allow this logic as well, i.e. `cumulative::Bool=false`. This way a lookback period could be used, *or* the current cumulative approach could be used.

As a cool extension, you might also consider adding another function call signature that allows passing a timestamp array (`Vector{TimeType}`) that might group by day and calculate intraday VWAP.

**Review by JTAmos — CHANGES_REQUESTED** [review]
Thanks for the proposed changes! These would be great things to add, just a few comments for things to take care of before merging.

File: src/ma.jl (hunk 4 above)
Comment (Bapi-Reddy) [reply src/ma.jl:407]: Got it I will update the function as per  the requirement

**Comment by JTAmos** [comment]
Also need to add the new functions to the exports in `src/Indicators.jl` and need to add the `Temporal.TS` wrappers to `src/temporal.jl`.

**Comment by JTAmos** [comment]
@Bapi-Reddy I went ahead and pushed some changes/fixes to this branch to resolve some of my comments. Let me know if you have any issues with the changes.

**Comment by JTAmos** [comment]
@Bapi-Reddy I'm going to merge the changes as they are now. If you are able to make the further suggested changes, just re-open another pull request and we can get those in as well.

**Comment by Bapi-Reddy** [comment]
Sure I will make the suggested changes with new pr.

**Pull request merged**